sentry/src/db/campaign.rs

```diff
@@ -49,6 +45,51 @@ pub async fn fetch_campaign(pool: DbPool, campaign: &Campaign) -> Result<Campaig
     Ok(Campaign::from(&row))
 }
 
+pub async fn get_campaigns_for_channel(pool: &DbPool, campaign: &Campaign) -> Result<Vec<Campaign>, PoolError> {
+    let client = pool.get().await?;
+    let statement = client.prepare("SELECT id, channel, creator, budget, validators, title, pricing_bounds, event_submission, ad_units, targeting_rules, created, active_from, active_to FROM campaigns WHERE channel_id = $1").await?;
+
+    let row = client.query(&statement, &[&campaign.channel.id()]).await?;
+
+    let campaigns = row.into_iter().map(|c| Campaign::from(&c)).collect();
```

Comment:
Should be: `rows.iter().map(Campaign::from).collect()`

`into_iter` will create an owned iterator (i.e. `|row: Row|` and for `Campaign::from` you need a `row: &Row` so no need for `into_iter()`, `iter()` is enough.

Since we call `iter()`, the `map()` gets a `row: &Row` which removes the need to wrap the `Campaign::from` in a closure all together, i.e. no need for `map(|row| Campaign::from(row))`